chain/client/src/client.rs

```diff
@@ -1373,8 +1373,19 @@ impl Client {
         self.chain.blocks_delay_tracker.mark_chunk_completed(&chunk_header, StaticClock::utc());
         self.block_production_info
             .record_chunk_collected(partial_chunk.height_created(), partial_chunk.shard_id());
+
         persist_chunk(partial_chunk, shard_chunk, self.chain.mut_chain_store())
             .expect("Could not persist chunk");
+
+        // We process chunk endorsements that were blocked by not having chunk complete.
+        self.chunk_endorsement_tracker.process_pending_endorsements(&chunk_header).expect(
```

Comment:
This expects in this method are a bit worrying. Not blocking for this PR but can you add a todo to add proper error handling here and add an item to do that to the tracking issue?